src/editor/codeeditor.cpp

```diff
@@ -13,13 +14,20 @@ CodeEditor::CodeEditor(QWidget *parent) : QPlainTextEdit(parent)
     setLineWrapMode(QPlainTextEdit::NoWrap);// don't move cursor to the next line where it's out of visible scope
 
     lineNumberArea = new LineNumberArea(this);
+    timer = new QTimer;
 
     //This signal is emitted when the text document needs an update of the specified rect.
     //If the text is scrolled, rect will cover the entire viewport area.
     //If the text is scrolled vertically, dy carries the amount of pixels the viewport was scrolled.
     connect(this, SIGNAL(updateRequest(QRect,int)), this, SLOT(updateLineNumberArea(QRect,int)));
-
+    connect(this, SIGNAL(textChanged()), this, SLOT(runLexer()));
     connect(this, SIGNAL(cursorPositionChanged()), this, SLOT(highlightCurrentLine()));
+    connect(timer, SIGNAL(timeout()), this, SLOT(saveStateInTheHistory()));
+    connect(this, SIGNAL(textChanged()), this, SLOT(changesAppeared()));
+
```

Comment:
In line 26 of code you connected signal with other signal

Reply:
That was signal I'll use in the future. Just forgot to delete it)